File: src/main/java/me/alegian/thaumcraft7/impl/common/menu/slot/T7ResultSlot.java
package me.alegian.thaumcraft7.impl.common.menu.slot;

import me.alegian.thaumcraft7.impl.common.menu.Menu;
import net.minecraft.world.Container;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.inventory.CraftingContainer;
import net.minecraft.world.inventory.ResultSlot;

/**
 * A Sized ResultSlot
 */
public class T7ResultSlot extends ResultSlot implements Sized {
  private final int size;
  private boolean mayPickup = true;

  public T7ResultSlot(Menu menu, CraftingContainer craftSlots, Container container, int id, int size) {
    super(menu.getPlayer(), craftSlots, container, id, menu.getSlotPose().getX(), menu.getSlotPose().getY());
    this.size = size;
  }

  @Override
  public int getSize() {
    return size;
  }

  public void setMayPickup(boolean mayPickup) {
    this.mayPickup = mayPickup;
  }

  @Override
  public boolean mayPickup(Player player) {
    return mayPickup;
  }
}

File: src/main/java/me/alegian/thavma/impl/common/menu/slot/T7ResultSlot.kt
package me.alegian.thavma.impl.common.menu.slot

import me.alegian.thavma.impl.common.menu.Menu
import net.minecraft.world.entity.player.Player
import net.minecraft.world.inventory.CraftingContainer
import net.minecraft.world.inventory.ResultContainer
import net.minecraft.world.inventory.ResultSlot
import net.minecraft.world.item.ItemStack

/**
 * A Sized ResultSlot, with a back-reference to a Menu
 */
open class T7ResultSlot<T : Menu>(override val menu: T, craftingContainer: CraftingContainer, container: ResultContainer, id: Int) : ResultSlot(menu.player, craftingContainer, container, id, 0, 0), DynamicSlot<T> {
  override var size = 0
  var mayPickup = true
  override var actualX = 0f
  override var actualY = 0f

  override fun onTake(player: Player, stack: ItemStack) {}

  override fun mayPickup(player: Player): Boolean {
    return this.mayPickup
  }
}
